### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by kiasha10**
[Feature/Merge] - LoginScreen

[Merge] feature/login → dev
[Feature] - LoginScreen

 ## Description
Added LoginScreen and Login Storyboard

## Checklist
Please review and check the following before submitting this pull request:
- [ ] The code follows the project's coding standards.
- [ ] 100% Unit test coverage for View Models

### CODE REVIEW

File: BlitzBoard/Features/LoginScreenViewController.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//
+import UIKit
+
+class LoginScreenViewController: UIViewController, UITextFieldDelegate {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+
```
Comment (sjacobs1) [inline BlitzBoard/Features/LoginScreenViewController.swift:14]: private outlets

File: BlitzBoard/Features/LoginScreenViewController.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    // MARK: IBActions
+    
+    @IBAction func loginButtonTapped(_ sender: Any) {
+        let email = emailTextField.text ?? ""
```
Comment (sjacobs1) [inline BlitzBoard/Features/LoginScreenViewController.swift]: private

File: BlitzBoard/Features/LoginScreenViewController.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    
+    @IBOutlet private weak var emailTextField: UITextField!
+    @IBOutlet private weak var passwordTextField: UITextField!
+    
+    // MARK: IBActions
+    
+    @IBAction private func loginButtonTapped(_ sender: Any) {
+        let email = emailTextField.text ?? ""
+        let password = passwordTextField.text ?? ""
+        if email == "kiashar@gmail.com" && password == "Kiasha1006" {
```
Comment (kieran-woodrow) [inline BlitzBoard/Features/LoginScreenViewController.swift]: this should be c`lling the authenticate function in your viewmodel

File: BlitzBoard/Features/LoginScreenViewController.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        emailTextField.delegate = self
+        passwordTextField.delegate = self
+    }
+    
+    // MARK: - Public Functions
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
```
Comment (kieran-woodrow) [inline BlitzBoard/Features/LoginScreenViewController.swift]: what does this do?

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: BlitzBoard/Features/LoginScreenViewController.swift (hunk 4 above)
Comment (kiasha10) [reply BlitzBoard/Features/LoginScreenViewController.swift]: I removed it now, I don't think it was necessary to have

File: BlitzBoard/Features/LoginScreenViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        guard let email, email.isEmpty,
+              let password, password.isEmpty else {
+            delegate?.loginResult(isSuccessful: false)
+            return false
+        }
+        
+        let isSuccessful = email == validEmail && password == validPassword
+        delegate?.loginResult(isSuccessful: isSuccessful)
+        
+        return email == validEmail && password == validPassword
```
Comment (kieran-woodrow) [inline BlitzBoard/Features/LoginScreenViewModel.swift]: you don't need to return anything. You re saying if the authentication is valid, then  delegate?.loginResult(isSuccessful: isSuccessful) which will take you the next screen. No need to  return email == validEmail && password == validPassword or have  -> Bool { in your function

can just be 
func authenticate(email: String?, password: String?) -> Bool {
        guard let email, email.isEmpty,
              let password, password.isEmpty else {
            delegate?.loginResult(isSuccessful: false)
            return false
        }

        let isSuccessful = email == validEmail && password == validPassword
        if isSuccessful {
        delegate?.loginResult(isSuccessful: isSuccessful)
        } else { 
        //handle your error here accordingly
        }

  
        }
        }

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: BlitzBoard/Features/LoginScreenViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    private weak var delegate: LoginViewModelDelegate?
+    
+    init(delegate: LoginViewModelDelegate) {
+        self.delegate = delegate
+    }
+    
+    // MARK: - Functions
+    
+    @discardableResult
+    func authenticate(email: String?, password: String?) -> Bool {
```
Comment (simekanidev) [inline BlitzBoard/Features/LoginScreenViewModel.swift]: Please remove @discardableResult

Comment (simekanidev) [inline BlitzBoard/Features/LoginScreenViewModel.swift]:
```suggestion
    func authenticate(email: String?, password: String?) {
```

File: BlitzBoard/Features/LoginScreenViewController.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    // MARK: Private Variables
+    
+    private lazy var viewModel = LoginViewModel(delegate: self)
+}
+
+    // MARK: Extensions
+
+extension LoginScreenViewController: LoginViewModelDelegate {
+    func loginResult(isSuccessful: Bool) {
+        if isSuccessful {
```
Comment (simekanidev) [inline BlitzBoard/Features/LoginScreenViewController.swift]: Rather handle this logic in the viewModel

**Review by kieran-woodrow — DISMISSED** [review]

**Review by AShozi — APPROVED** [review]

**Review by simekanidev — APPROVED** [review]

**Pull request merged**